enhancements/vmaas/README.md

```diff
@@ -0,0 +1,472 @@
+---
+title: Virtualization as a service
+authors:
+  - Adrien Gentil
+creation-date: 2025-09-15
+last-updated: 2025-09-15
+tracking-link:
+  - TBD
+see-also:
+replaces:
+superseded-by:
+---
+
+# Virtualization as a service
+
+
+## Summary
+
+We want to provide a service that allows a tenant to create and manage VMs.
+
+## Motivation
+
+TBD
+
+### User Stories
+
+#### As a tenant
+
+- I want to list a pre-defined VM templates
+- I want to create a VM based on a pre-defined template
+- I want a VM that has access to specialized hardware (e.g.: GPU)
+- I want to customize the pre-defined template for my usage
+- I want to attach and detach block storage on my VM that transcends the lifecycle of the VM
+- I want to manage the lifecycle of my VM (start/stop/terminate)
+- I want to import my OS base image
+- I want to list available base OS images
+- I want to manage (create, list, delete) my private network
+- I want to attach my VMs in a virtual private network
+- I want my VM to be exposed outside of my private network
+- I want my VMs to be attached to my own physical L2 network
+- I want to connect on my VM through serial console
+- I want to keep running my VM when a node goes out of service
+
+#### As a service provider:
+
+- I want to define VM templates
+- I want to manually trigger VM migration
+
+### Goals
+
+TBD
+
+### Non-Goals
+
+TBD
+
+## Proposal
+
+### Network isolation
+
+This design relies on UDN (User Defined Networking, it is the networking component provided with Openshift, it allows the definition of virtual private subnets, and provides the ability to connect on physical networks (localnet).
+
+Moverover, Kubevirt supports UDN layer 2 mode which offers the ability to live-migrate VMs across OCP nodes.
+
+### Ingress
+
+We can expose services running on VM(s) using a “LoadBalancer” service (e.g.: relying on MetalLB).
+
+### APIs
+
+Virtual private subnet creation:
+
+```
+apiVersion: cloudkit.openshift.io/v1alpha1
+kind: VirtualPrivateSubnet
```

Comment:
Seems this should be in a separate proposal about network isolation.

Reply:
Yes, as discussed, networking won't be part of this story